tls/s2n_server_extensions.c

```diff
@@ -177,6 +188,9 @@ int s2n_server_extensions_recv(struct s2n_connection *conn, struct s2n_blob *ext
         case TLS_EXTENSION_SESSION_TICKET:
             GUARD(s2n_recv_server_session_ticket_ext(conn, &extension));
             break;
+        case TLS_EXTENSION_SUPPORTED_VERSIONS:
+            GUARD(s2n_extensions_server_supported_versions_recv(conn, &extension));
```

Comment:
Non blocking, but we may want to consider changing our extension parsing logic(on both client and server) to search for this extension first.

Implementing processing for all the other 1.3 extensions will be less error prone if we can assume version negotiation is already complete when their handlers are called. For example, we don't want to take action on the Key Share extension if we aren't sure we're negotiating 1.3 with the client.

Reply:
+1 on this point. i think using `s2n_client_hello_get_extension()` in the handshake would be the right way to go about it but it would require more changes to the hello and extension parsing code.

Reply:
Issue created here to change parsing logic https://github.com/awslabs/s2n/issues/1173